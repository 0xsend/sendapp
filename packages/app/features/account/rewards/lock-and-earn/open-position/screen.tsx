import { Button, Card, H1, Paragraph, XStack, YStack, SubmitButton, Stack } from '@my/ui'
import { z } from 'zod'
import { SchemaForm, formFields } from 'app/utils/SchemaForm'
import { FormProvider, useForm } from 'react-hook-form'
import type { PropsWithChildren } from 'react'
import { coins, coinsDict, sendCoin } from 'app/data/coins'
import { IconCoin } from 'app/components/icons/IconCoin'
import { useSendAccountBalances } from 'app/utils/useSendAccountBalances'
import { formatUnits } from 'viem'
import formatAmount from 'app/utils/formatAmount'

const removeDuplicateInString = (text: string, substring: string) => {
  const [first, ...after] = text.split(substring)
  return first + (after.length ? `${substring}${after.join('')}` : '')
}

export function OpenPosition() {
  return (
    <YStack pt={'$size.3.5'} $gtLg={{ pt: 0 }} f={1} $gtMd={{ ml: '$4' }}>
      <YStack pb={'$size.3.5'} gap="$size.0.9">
        <H1 size={'$9'} fontWeight={'900'} color="$color12" tt={'uppercase'}>
          Open a new position
        </H1>
        <Paragraph color={'$color10'} size={'$5'} maw={650}>
          Liquidity Providers get to share 1.5% of all swaps (added to your rewards automatically),
          and 5b in $SEND rewards over 12 months.
        </Paragraph>
      </YStack>

      <YStack $gtMd={{ fd: 'row' }} gap={'$size.1.5'}>
        <OpenPositionForm />
      </YStack>
    </YStack>
  )
}

const OpenPositionForm = () => {
  const form = useForm<z.infer<typeof OpenPositionSchema>>()
  const { balances } = useSendAccountBalances()

  const formToken = form.watch('token')
  const formTokenAmount = form.watch('tokenAmount')
  const formSendAmount = form.watch('sendAmount')

  const [tokenSymbol, tokenDecimals] = [
    coinsDict[formToken]?.symbol,
    coinsDict[formToken]?.decimals,
  ]

  const sendBalance = (
    balances?.SEND ? formatUnits(balances.SEND, sendCoin.decimals) : 0n
  ).toString()
  const tokenBalance = formatUnits(balances?.[tokenSymbol] ?? 0n, tokenDecimals)

  const canSubmit =
    formTokenAmount &&
    Number(formTokenAmount) > 0 &&
    Number(formTokenAmount) <= Number(tokenBalance) &&
    formSendAmount &&
    Number(formSendAmount) > 0 &&
    Number(formSendAmount) <= Number(sendBalance)

  return (
    <FormProvider {...form}>
      <SchemaForm
        form={form}
        schema={OpenPositionSchema}
        onSubmit={() => {}}
        props={{
          tokenAmount: {
            ...sharedStyles,
            onChangeText: (text) => {
              const formattedText = removeDuplicateInString(text, '.').replace(/[^0-9.]/g, '') //remove duplicate "." then filter out any letters
              form.setValue('tokenAmount', formattedText)
            },
          },
          sendAmount: {
            ...sharedStyles,
            inputMode: 'numeric',
            onChangeText: (text) => {
              const formattedText = text.replace(/[^0-9]/g, '') // only accept numbers
              form.setValue('sendAmount', formattedText)
            },
          },
          token: {
            supportedCoins: coins.filter(({ symbol }) => symbol !== 'SEND'),
          },
        }}
        defaultValues={{
          token: 'eth',
        }}
        formProps={{
          testID: 'OpenPositionForm',
          $gtSm: { maxWidth: '100%' },
          jc: 'flex-start',
          als: 'flex-start',
          f: 1,
          height: '100%',
        }}
        renderAfter={({ submit }) => (
          <XStack mt={'$size.3.5'}>
            <SubmitButton
              variant={!canSubmit ? 'outlined' : undefined}
              theme={canSubmit ? 'green' : 'dim'}
              borderRadius={'$4'}
              disabled={!canSubmit}
              // @TODO: submit
              onPress={() => console.log('press')}
            >
              <Button.Text
                tt={'uppercase'}
                color={canSubmit ? '$color1' : '$color12'}
                fontWeight={500}
                ff="$mono"
              >
                {canSubmit ? 'Provide Liquidity' : 'Enter USDC or SEND Amount'}
              </Button.Text>
            </SubmitButton>
          </XStack>
        )}
      >
        {({ token, tokenAmount, sendAmount }) => (
          <YStack $gtMd={{ fd: 'row' }} gap={'$size.1.5'}>
            <FormCard
              balance={tokenBalance}
              onMaxPress={() => form.setValue('tokenAmount', tokenBalance)}
            >
              <Stack w={'$size.3.5'}>
                {coinsDict[formToken] && (
                  <IconCoin coin={coinsDict[formToken]} size={'$size.3.5'} />
                )}
              </Stack>
              {tokenAmount}
              {token}
            </FormCard>

            <FormCard
              balance={sendBalance}
              onMaxPress={() => form.setValue('sendAmount', sendBalance)}
            >
              <Stack>
                <IconCoin coin={sendCoin} size={'$size.3.5'} />
              </Stack>
              {sendAmount}
              <Paragraph fontWeight={600} fontSize={'$8'}>
                SEND
              </Paragraph>
            </FormCard>
          </YStack>
        )}
      </SchemaForm>
    </FormProvider>
  )
}

const OpenPositionSchema = z.object({
  token: formFields.coin,
  tokenAmount: formFields.text,
  sendAmount: formFields.text,
})

const FormCard = ({
  balance,
  onMaxPress,
  children,
}: {
  balance: string
  onMaxPress: () => void
} & PropsWithChildren) => {
  return (
    <Card p={'$size.3.5'} flex={1}>
      <XStack
        pb="$size.0.9"
        borderBottomColor={'$color10'}
        borderBottomWidth={1}
        justifyContent="space-between"
        alignItems="center"
        gap="$size.0.9"
      >
        {children}
      </XStack>

      <XStack pt="$size.0.9" justifyContent="space-between">
        <XStack gap="$size.0.75">
          <Paragraph color="$color10" size={'$5'}>
            Balance:
          </Paragraph>
          <Paragraph size={'$5'} fontWeight={'600'}>
            {formatAmount(balance, 9, 2)}
          </Paragraph>
        </XStack>
        <Button unstyled onPress={onMaxPress}>
          <Button.Text size={'$5'} color="$color10" textDecorationLine="underline">
            MAX
          </Button.Text>
        </Button>
      </XStack>
    </Card>
  )
}

const sharedStyles = {
  fieldsetProps: { flex: 1 },
  backgroundColor: '$color1',
  flex: 1,
  inputMode: 'decimal',
  fontSize: '$9',
  ff: '$mono',
  placeholder: '0',
  p: 0,
  lineHeight: '$1',
  borderColor: 'transparent',
  outlineColor: 'transparent',
  fontStyle: 'normal',
  '$theme-dark': {
    placeholderTextColor: '$color10',
  },
  $sm: {
    w: '100%',
  },
} as const
