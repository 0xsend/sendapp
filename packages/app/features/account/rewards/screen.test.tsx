import { Wrapper } from 'app/utils/__mocks__/Wrapper'
import { RewardsScreen } from './screen'
import { render } from '@testing-library/react-native'

jest.mock('app/utils/useUser')
jest.mock('app/utils/distributions', () => ({
  useDistributions: () => ({
    data: [
      {
        number: 1,
        chain_id: 845337,
        qualification_end: Date.UTC(2024, 6, 15),
        distribution_shares: [
          {
            amount: 1,
            index: 1,
          },
        ],
        distribution_verifications_summary: [
          {
            tag_referrals: 123,
          },
        ],
      },
    ],
  }),
  useSendMerkleDropTrancheActive: jest.fn().mockReturnValue({
    data: true,
    isSuccess: true,
    error: null,
  }),
  useSendMerkleDropIsClaimed: jest.fn().mockReturnValue({
    data: true,
    isSuccess: true,
    error: null,
  }),
  usePrepareSendMerkleDropClaimTrancheWrite: jest.fn().mockReturnValue({
    data: {},
    isSuccess: true,
    error: null,
  }),
}))

jest.mock('app/utils/useChainAddresses', () => ({
  useChainAddresses: jest.fn().mockReturnValue({ data: { address: '0x123' } }),
}))
jest.mock('app/routers/params', () => ({
  useDistributionNumberParams: () => ({ distributionNumber: 1 }),
  useDistributionNumber: () => [1, jest.fn()],
}))
jest.mock('wagmi')
jest.mock('@rainbow-me/rainbowkit', () => ({
  useConnectModal: jest.fn().mockReturnValue({ openConnectModal: jest.fn() }),
}))
jest.mock('@my/wagmi', () => ({
  __esModule: true,
  ...jest.requireActual('@my/wagmi'),
  baseMainnetClient: {
    chain: {
      id: 845337,
    },
    simulateContract: jest.fn().mockResolvedValue({}),
  },
  baseMainnetBundlerClient: {
    sendUserOperation: jest.fn(),
    waitForUserOperationReceipt: jest.fn().mockResolvedValue({ success: true }),
  },
  useReadSendTokenBalanceOf: jest.fn().mockReturnValue({
    data: 0n,
    isSuccess: true,
    error: null,
  }),
}))
jest.mock('app/utils/coin-gecko', () => ({
  useSendPrice: jest
    .fn()
    .mockReturnValue({ data: { 'send-token': { usd: 1 } }, isSuccess: true, error: null }),
}))
jest.mock('app/utils/tags', () => ({
  useConfirmedTags: jest.fn().mockReturnValue({ data: [{ name: 'tag1' }, { name: 'tag2' }] }),
}))
describe('EarnTokensScreen', () => {
  it('renders', () => {
    jest.useFakeTimers()
    jest.setSystemTime(Date.UTC(2024, 6, 12))
    const tree = render(
      <Wrapper>
        <RewardsScreen />
      </Wrapper>
    )

    expect(tree.toJSON()).toMatchSnapshot('EarnTokensScreen')
  })
})
