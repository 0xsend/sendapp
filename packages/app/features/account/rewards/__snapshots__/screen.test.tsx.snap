// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EarnTokensScreen renders: EarnTokensScreen 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "column",
      "gap": 32,
      "height": "95%",
      "justifyContent": "space-between",
      "marginBottom": "auto",
      "marginTop": "auto",
      "paddingBottom": 7,
    }
  }
>
  <View
    style={
      {
        "flex": 2,
        "flexDirection": "column",
        "gap": 18,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
          "gap": 24,
          "paddingTop": 32,
        }
      }
    >
      <View
        style={
          {
            "gap": 7,
          }
        }
      >
        <Text
          cancelable={true}
          disabled={false}
          focusable={true}
          id=":r0:"
          minPressDuration={0}
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onPress={[Function]}
          onPressIn={[Function]}
          onPressOut={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "color": "hsla(0, 0%, 100%, 1)",
              "display": "flex",
              "fontFamily": "System",
              "fontSize": 19.2,
              "fontWeight": "400",
              "lineHeight": 19,
            }
          }
          suppressHighlighting={true}
          userSelect="none"
        >
          ROUND
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-around",
              "marginTop": "auto",
              "width": "100%",
            }
          }
        >
          <View>
            <Text
              accessibilityRole="header"
              selectable={true}
              style={
                {
                  "color": "hsla(0, 0%, 100%, 1)",
                  "fontFamily": "System",
                  "fontSize": 54,
                  "fontWeight": "300",
                  "lineHeight": 51,
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                }
              }
              suppressHighlighting={true}
            >
              #
              1
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "stretch",
                "borderBottomColor": "#3E4A3C",
                "borderLeftColor": "#3E4A3C",
                "borderRightColor": "#3E4A3C",
                "borderRightWidth": 1,
                "borderStyle": "solid",
                "borderTopColor": "#3E4A3C",
                "height": "100%",
                "marginLeft": 18,
                "marginRight": 18,
                "width": 0,
              }
            }
          />
          <View
            style={
              {
                "flex": 1,
                "flexDirection": "column",
                "gap": 18,
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "column",
                  "gap": 7,
                  "justifyContent": "center",
                  "maxWidth": 312,
                }
              }
            >
              <Text
                selectable={true}
                style={
                  {
                    "color": "hsla(0, 0%, 100%, 1)",
                    "fontFamily": "System",
                    "fontSize": 19.2,
                    "fontWeight": "400",
                    "lineHeight": 19,
                  }
                }
                suppressHighlighting={true}
              >
                Expired Jul 12, 2024
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "column",
                }
              }
            >
              <Text
                cancelable={true}
                disabled={false}
                focusable={true}
                id=":r1:"
                minPressDuration={0}
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onPress={[Function]}
                onPressIn={[Function]}
                onPressOut={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "color": "hsla(0, 0%, 100%, 1)",
                    "display": "flex",
                    "fontFamily": "System",
                    "fontSize": 16.8,
                    "fontWeight": "400",
                    "lineHeight": 19,
                  }
                }
                suppressHighlighting={true}
                userSelect="none"
              >
                Status
              </Text>
              <Text
                accessibilityRole="header"
                selectable={true}
                style={
                  {
                    "color": "hsla(0, 0%, 100%, 1)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "lineHeight": 27,
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 0,
                    "textTransform": "none",
                  }
                }
                suppressHighlighting={true}
              >
                OPEN
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
            "gap": 7,
            "marginBottom": "auto",
            "marginTop": "auto",
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "column",
              "gap": 7,
            }
          }
        >
          <View
            style={
              {
                "flex": 1,
                "gap": 7,
              }
            }
          >
            <View
              style={
                {
                  "backgroundColor": "transparent",
                  "borderBottomLeftRadius": 12,
                  "borderBottomRightRadius": 12,
                  "borderBottomWidth": 1,
                  "borderLeftWidth": 1,
                  "borderRightWidth": 1,
                  "borderStyle": "solid",
                  "borderTopLeftRadius": 12,
                  "borderTopRightRadius": 12,
                  "borderTopWidth": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "paddingBottom": 18,
                  "paddingLeft": 18,
                  "paddingRight": 18,
                  "paddingTop": 18,
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "column",
                    "gap": 7,
                  }
                }
              >
                <Text
                  cancelable={true}
                  disabled={false}
                  focusable={true}
                  id=":r2:"
                  minPressDuration={0}
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onPress={[Function]}
                  onPressIn={[Function]}
                  onPressOut={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "color": "hsla(112, 22%, 59%, 1)",
                      "display": "flex",
                      "fontFamily": "System",
                      "fontSize": 19.2,
                      "fontWeight": "400",
                      "lineHeight": 19,
                    }
                  }
                  suppressHighlighting={true}
                  userSelect="none"
                >
                  Snapshot Send Balance
                </Text>
                <Text
                  selectable={true}
                  style={
                    {
                      "color": "hsla(0, 0%, 100%, 1)",
                      "fontFamily": "System",
                      "fontSize": 24,
                      "fontWeight": "500",
                      "lineHeight": 19,
                    }
                  }
                  suppressHighlighting={true}
                >
                  0 SEND
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "flex": 1,
                "flexDirection": "row",
                "gap": 7,
              }
            }
          >
            <View
              style={
                {
                  "backgroundColor": "transparent",
                  "borderBottomLeftRadius": 12,
                  "borderBottomRightRadius": 12,
                  "borderBottomWidth": 1,
                  "borderLeftWidth": 1,
                  "borderRightWidth": 1,
                  "borderStyle": "solid",
                  "borderTopLeftRadius": 12,
                  "borderTopRightRadius": 12,
                  "borderTopWidth": 1,
                  "flex": 2,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "paddingBottom": 18,
                  "paddingLeft": 18,
                  "paddingRight": 18,
                  "paddingTop": 18,
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "column",
                    "gap": 7,
                  }
                }
              >
                <Text
                  cancelable={true}
                  disabled={false}
                  focusable={true}
                  id=":r3:"
                  minPressDuration={0}
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onPress={[Function]}
                  onPressIn={[Function]}
                  onPressOut={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "color": "hsla(112, 22%, 59%, 1)",
                      "display": "flex",
                      "fontFamily": "System",
                      "fontSize": 19.2,
                      "fontWeight": "400",
                      "lineHeight": 19,
                    }
                  }
                  suppressHighlighting={true}
                  userSelect="none"
                >
                  Min Balance required
                </Text>
                <Text
                  selectable={true}
                  style={
                    {
                      "color": "hsla(0, 0%, 100%, 1)",
                      "fontFamily": "System",
                      "fontSize": 24,
                      "fontWeight": "500",
                      "lineHeight": 19,
                    }
                  }
                  suppressHighlighting={true}
                >
                  ?
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "backgroundColor": "transparent",
                  "borderBottomLeftRadius": 12,
                  "borderBottomRightRadius": 12,
                  "borderBottomWidth": 1,
                  "borderLeftWidth": 1,
                  "borderRightWidth": 1,
                  "borderStyle": "solid",
                  "borderTopLeftRadius": 12,
                  "borderTopRightRadius": 12,
                  "borderTopWidth": 1,
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "paddingBottom": 18,
                  "paddingLeft": 18,
                  "paddingRight": 18,
                  "paddingTop": 18,
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "column",
                    "gap": 7,
                  }
                }
              >
                <Text
                  cancelable={true}
                  disabled={false}
                  focusable={true}
                  id=":r4:"
                  minPressDuration={0}
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onPress={[Function]}
                  onPressIn={[Function]}
                  onPressOut={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "color": "hsla(112, 22%, 59%, 1)",
                      "display": "flex",
                      "fontFamily": "System",
                      "fontSize": 19.2,
                      "fontWeight": "400",
                      "lineHeight": 19,
                    }
                  }
                  suppressHighlighting={true}
                  userSelect="none"
                >
                  Referrals
                </Text>
                <Text
                  selectable={true}
                  style={
                    {
                      "color": "hsla(0, 0%, 100%, 1)",
                      "fontFamily": "System",
                      "fontSize": 24,
                      "fontWeight": "500",
                      "lineHeight": 19,
                    }
                  }
                  suppressHighlighting={true}
                >
                  123
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "hsla(191, 32%, 10%, 1)",
                "borderBottomLeftRadius": 12,
                "borderBottomRightRadius": 12,
                "borderTopLeftRadius": 12,
                "borderTopRightRadius": 12,
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
                "minHeight": 198,
                "paddingBottom": 32,
                "paddingLeft": 32,
                "paddingRight": 32,
                "paddingTop": 32,
                "position": "relative",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "gap": 18,
                  "justifyContent": "center",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                }
              }
            >
              <View
                style={
                  {
                    "gap": 32,
                  }
                }
              >
                <Text
                  cancelable={true}
                  disabled={false}
                  focusable={true}
                  id=":r5:"
                  minPressDuration={0}
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onPress={[Function]}
                  onPressIn={[Function]}
                  onPressOut={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "color": "#86AE80",
                      "display": "flex",
                      "fontFamily": "System",
                      "fontSize": 19.2,
                      "fontWeight": "400",
                      "lineHeight": 19,
                      "textAlign": "left",
                    }
                  }
                  suppressHighlighting={true}
                  userSelect="none"
                >
                  Rewards
                </Text>
                <View>
                  <Text
                    selectable={true}
                    style={
                      {
                        "color": "hsla(191, 32%, 10%, 1)",
                        "fontFamily": "System",
                        "fontSize": 55.199999999999996,
                        "fontWeight": "500",
                        "lineHeight": 40,
                      }
                    }
                    suppressHighlighting={true}
                  >
                    1 SEND
                  </Text>
                  <Text
                    selectable={true}
                    style={
                      {
                        "color": "hsla(112, 22%, 50%, 1)",
                        "fontFamily": "System",
                        "fontSize": 16.8,
                        "fontWeight": "400",
                        "lineHeight": 19,
                        "opacity": 0.6,
                      }
                    }
                    suppressHighlighting={true}
                  >
                    1.00 USD
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "marginBottom": "auto",
        "marginTop": "auto",
      }
    }
  >
    <RCTScrollView
      scrollEnabled={true}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      style={
        {
          "marginBottom": "auto",
          "marginTop": "auto",
          "overflow": "scroll",
          "paddingBottom": 24,
        }
      }
    >
      <View>
        <View
          style={
            {
              "flexDirection": "row",
              "gap": 7,
              "justifyContent": "space-between",
              "marginLeft": "auto",
              "marginRight": "auto",
              "maxWidth": 1072,
              "width": "100%",
            }
          }
        >
          <View
            disabled={true}
            style={
              {
                "backgroundColor": "hsla(191, 32%, 10%, 1)",
                "borderBottomLeftRadius": 6,
                "borderBottomRightRadius": 6,
                "borderTopLeftRadius": 6,
                "borderTopRightRadius": 6,
                "flex": 1,
                "flexDirection": "column",
                "height": 28,
                "justifyContent": "center",
                "maxWidth": 84,
                "minWidth": 74,
                "opacity": 0.5,
                "position": "relative",
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "hsla(112, 22%, 59%, 1)",
                  "fontFamily": "System",
                  "fontSize": 13.2,
                  "fontWeight": "300",
                  "lineHeight": 16,
                  "textAlign": "center",
                }
              }
              suppressHighlighting={true}
            >
              # 1
            </Text>
          </View>
          <View
            disabled={true}
            style={
              {
                "backgroundColor": "hsla(191, 32%, 10%, 1)",
                "borderBottomLeftRadius": 6,
                "borderBottomRightRadius": 6,
                "borderTopLeftRadius": 6,
                "borderTopRightRadius": 6,
                "flex": 1,
                "flexDirection": "column",
                "height": 28,
                "justifyContent": "center",
                "maxWidth": 84,
                "minWidth": 74,
                "opacity": 0.5,
                "position": "relative",
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "hsla(112, 22%, 59%, 1)",
                  "fontFamily": "System",
                  "fontSize": 13.2,
                  "fontWeight": "300",
                  "lineHeight": 16,
                  "textAlign": "center",
                }
              }
              suppressHighlighting={true}
            >
              # 2
            </Text>
          </View>
          <View
            disabled={true}
            style={
              {
                "backgroundColor": "hsla(191, 32%, 10%, 1)",
                "borderBottomLeftRadius": 6,
                "borderBottomRightRadius": 6,
                "borderTopLeftRadius": 6,
                "borderTopRightRadius": 6,
                "flex": 1,
                "flexDirection": "column",
                "height": 28,
                "justifyContent": "center",
                "maxWidth": 84,
                "minWidth": 74,
                "opacity": 0.5,
                "position": "relative",
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "hsla(112, 22%, 59%, 1)",
                  "fontFamily": "System",
                  "fontSize": 13.2,
                  "fontWeight": "300",
                  "lineHeight": 16,
                  "textAlign": "center",
                }
              }
              suppressHighlighting={true}
            >
              # 3
            </Text>
          </View>
          <View
            disabled={true}
            style={
              {
                "backgroundColor": "hsla(191, 32%, 10%, 1)",
                "borderBottomLeftRadius": 6,
                "borderBottomRightRadius": 6,
                "borderTopLeftRadius": 6,
                "borderTopRightRadius": 6,
                "flex": 1,
                "flexDirection": "column",
                "height": 28,
                "justifyContent": "center",
                "maxWidth": 84,
                "minWidth": 74,
                "opacity": 0.5,
                "position": "relative",
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "hsla(112, 22%, 59%, 1)",
                  "fontFamily": "System",
                  "fontSize": 13.2,
                  "fontWeight": "300",
                  "lineHeight": 16,
                  "textAlign": "center",
                }
              }
              suppressHighlighting={true}
            >
              # 4
            </Text>
          </View>
          <View
            disabled={true}
            style={
              {
                "backgroundColor": "hsla(191, 32%, 10%, 1)",
                "borderBottomLeftRadius": 6,
                "borderBottomRightRadius": 6,
                "borderTopLeftRadius": 6,
                "borderTopRightRadius": 6,
                "flex": 1,
                "flexDirection": "column",
                "height": 28,
                "justifyContent": "center",
                "maxWidth": 84,
                "minWidth": 74,
                "opacity": 0.5,
                "position": "relative",
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "hsla(112, 22%, 59%, 1)",
                  "fontFamily": "System",
                  "fontSize": 13.2,
                  "fontWeight": "300",
                  "lineHeight": 16,
                  "textAlign": "center",
                }
              }
              suppressHighlighting={true}
            >
              # 5
            </Text>
          </View>
          <View
            disabled={true}
            style={
              {
                "backgroundColor": "hsla(191, 32%, 10%, 1)",
                "borderBottomLeftRadius": 6,
                "borderBottomRightRadius": 6,
                "borderTopLeftRadius": 6,
                "borderTopRightRadius": 6,
                "flex": 1,
                "flexDirection": "column",
                "height": 28,
                "justifyContent": "center",
                "maxWidth": 84,
                "minWidth": 74,
                "opacity": 0.5,
                "position": "relative",
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "hsla(112, 22%, 59%, 1)",
                  "fontFamily": "System",
                  "fontSize": 13.2,
                  "fontWeight": "300",
                  "lineHeight": 16,
                  "textAlign": "center",
                }
              }
              suppressHighlighting={true}
            >
              # 6
            </Text>
          </View>
          <View
            disabled={true}
            style={
              {
                "backgroundColor": "hsla(191, 32%, 10%, 1)",
                "borderBottomLeftRadius": 6,
                "borderBottomRightRadius": 6,
                "borderTopLeftRadius": 6,
                "borderTopRightRadius": 6,
                "flex": 1,
                "flexDirection": "column",
                "height": 28,
                "justifyContent": "center",
                "maxWidth": 84,
                "minWidth": 74,
                "opacity": 0.5,
                "position": "relative",
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "hsla(112, 22%, 59%, 1)",
                  "fontFamily": "System",
                  "fontSize": 13.2,
                  "fontWeight": "300",
                  "lineHeight": 16,
                  "textAlign": "center",
                }
              }
              suppressHighlighting={true}
            >
              # 7
            </Text>
          </View>
          <View
            disabled={true}
            style={
              {
                "backgroundColor": "hsla(191, 32%, 10%, 1)",
                "borderBottomLeftRadius": 6,
                "borderBottomRightRadius": 6,
                "borderTopLeftRadius": 6,
                "borderTopRightRadius": 6,
                "flex": 1,
                "flexDirection": "column",
                "height": 28,
                "justifyContent": "center",
                "maxWidth": 84,
                "minWidth": 74,
                "opacity": 0.5,
                "position": "relative",
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "hsla(112, 22%, 59%, 1)",
                  "fontFamily": "System",
                  "fontSize": 13.2,
                  "fontWeight": "300",
                  "lineHeight": 16,
                  "textAlign": "center",
                }
              }
              suppressHighlighting={true}
            >
              # 8
            </Text>
          </View>
          <View
            disabled={true}
            style={
              {
                "backgroundColor": "hsla(191, 32%, 10%, 1)",
                "borderBottomLeftRadius": 6,
                "borderBottomRightRadius": 6,
                "borderTopLeftRadius": 6,
                "borderTopRightRadius": 6,
                "flex": 1,
                "flexDirection": "column",
                "height": 28,
                "justifyContent": "center",
                "maxWidth": 84,
                "minWidth": 74,
                "opacity": 0.5,
                "position": "relative",
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "hsla(112, 22%, 59%, 1)",
                  "fontFamily": "System",
                  "fontSize": 13.2,
                  "fontWeight": "300",
                  "lineHeight": 16,
                  "textAlign": "center",
                }
              }
              suppressHighlighting={true}
            >
              # 9
            </Text>
          </View>
          <View
            disabled={true}
            style={
              {
                "backgroundColor": "hsla(191, 32%, 10%, 1)",
                "borderBottomLeftRadius": 6,
                "borderBottomRightRadius": 6,
                "borderTopLeftRadius": 6,
                "borderTopRightRadius": 6,
                "flex": 1,
                "flexDirection": "column",
                "height": 28,
                "justifyContent": "center",
                "maxWidth": 84,
                "minWidth": 74,
                "opacity": 0.5,
                "position": "relative",
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "hsla(112, 22%, 59%, 1)",
                  "fontFamily": "System",
                  "fontSize": 13.2,
                  "fontWeight": "300",
                  "lineHeight": 16,
                  "textAlign": "center",
                }
              }
              suppressHighlighting={true}
            >
              # 10
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
